pkg/controller/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -1035,9 +1055,9 @@ func assertToolchainStatusNotificationCreated(t *testing.T, fakeClient *test.Fak
 	return &notifications.Items[0]
 }
 
-func assertToolchainStatusNotificationNotCreated(t *testing.T, fakeClient *test.FakeClient) {
+func assertToolchainStatusNotificationNotCreated(t *testing.T, fakeClient *test.FakeClient, notificationType string) {
 	var notification toolchainv1alpha1.Notification
-	err := fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs, "toolchainstatus-unready"),
+	err := fakeClient.Get(context.Background(), test.NamespacedName(test.HostOperatorNs, notificationType),
```

Comment:
It seems that we actually had a bug here. We can't just check if there is a notification with such name (== notificationType) because the actual notification resource name is in `<notificationType>-<timestamp>` format.
Instead of using `Get` we have to use `List` to list all the notifications and check their names if there any notification with the name which start with: `<notificationType>-`. There should not be any notification with such a name.
Otherwise this assertion will always pass.